src/Orc.Controls.Tests/Controls/SaveFilePicker/SaveFilePickerFacts.cs

```diff
@@ -0,0 +1,80 @@
+﻿namespace Orc.Controls.Tests.Controls
+{
+    using System.Threading.Tasks;
+    using Catel.Services;
+    using Moq;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class SaveFilePickerFacts
+    {
+        [TestCase(null, null)]
+        [TestCase("some_item.txt", "some_item.txt")]
+        public async Task UsesDefaultFileName_WithoutFileName_Async(string input, string expectedOutput)
```

Comment:
Unclear name without knowing details SaveFileService implementation
The service checks the 'determine context' called
The name assumes this is a sort of check for fallback value
So without getting into context is hard to understand what's the link between